sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -610,13 +610,16 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
       inputTable: Table,
       filePath: String,
       clippedSchema: MessageType): Table = {
+    convertToUnsigned(convertToDecimal(inputTable, filePath, clippedSchema))
```

Comment:
Could we do the two castings in one iterating ?

Reply:
The logic is already a little complicated, if not decreases performance I think it's acceptable.

Reply:
If I understand correctly, the only diff is type casting for different types. The original is for decimal and yours is for unsigned integral. Others are simliar.